### ISSUE TITLE
Azure Files share creation fails with 403 error

### ISSUE THREAD

**Issue by doherty100** [Azure-Samples/azuresandbox#29]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [  X ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Install sandbox v2.9.0
> Update azurerm to v3.98.0
> Apply vnet-app configuration

### Any log messages given by the failure
> `Error: checking for existing File Share "myfileshare" (Account "Account \"staymunn1rch5rh\" (IsEdgeZone false / ZoneName \"\" / Subdomain Type \"file\" / DomainSuffix \"core.windows.net\")"): executing request: unexpected status 403 (403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.) with response: <?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.`

### Expected/desired behavior
> Configuration should provision successfully

### OS and Version?
> N/A

### Versions
> AzureSandbox v2.9.0 + azurerm v3.98.0

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by doherty100** [comment]
Opened https://github.com/hashicorp/terraform-provider-azurerm/issues/25527

**Comment by doherty100** [comment]
Resolved in azurerm v3.100.0.
